src/services/item/plugins/invitation/specifications.md

```diff
@@ -0,0 +1,101 @@
+# Invitations
+
+This plugin handles invitations.
+
+## Available routes
+
+### Get a single invitation
+
+Route: `GET /items/invitations/:id` 
+Params: 
+> Does not require authentication
+
+### Create invitations for an item
+
+Route: `POST /items/:id/invite`
+
+### Add users from a CSV
+
+Route: `POST /items/:id/invitations/upload-csv`
+Body: Multipart Form containing a csv file.
+
+#### CSV File specifications
+
+| Column Name | Description | Required | Default Value |
+|---|---|---|---|
+| `email` | Email adress of the user | Yes | None (required) |
+| `name` | Name to use when creating the user | No | Use the name in the email |
+| `permission` | Permission level you want to grant to the user | No | `read` |
+| `group_name` | The group to which the user should be added. Only relevant when using the template feature | If present should be present for all users | None (required if present for one user) | 
+
+**NOTES**: 
+- Order of the columns is not important.
+- Spaces around the delimiter are supported
+
+#### Valid CSV files
+
+A single user specified only by email, will receive `read` permission
+```csv
+email
+alice@example.com
+```
+
+Multiple users specified with email and name
+```csv
+name, email
+Alice, alice@example.com
+Bob, bob@example.com
+```
+
+Also works with no spaces
+```csv
+name,email
+Alice,alice@example.com
+Bob,bob@example.com
+```
+
+Multiple users with not all permissions set
+```csv
+email, name, permission
+alice@example.com, Alice, read
+bob@example.com, Bob
+```
+
+Permissions can be left blank too
+```csv
+email, permission, name
+alice@example.com,, Alice
+```
+
+#### Handling duplicates
+
+#### Memberships
+
+A data entry targeting an already existing membership will result in a `Modifying Existing` error.
+
+#### Invitations
+
+An email that is already invited on an item will be droped. 
```

Comment:
So if I understand correctly, you can set an invitation above or under, but never on the same item? can't we allow it as long as it does not break the permission rules? I can be postpone to later and saved in an issue.

Reply:
Yes you are right, you can set an invitation above or under an invitation that already exists, but not exactly on the same item for the same user. This is a mirror of how the memberships behave currently (not ideal). 
Because if you can set it on the same item, then you need to check if it is more, or less restrictive. 

I am aware that you can potentially introduce inconsistent invitations with this implementation, so this needs to be checked. Maybe it would cover the same item use-case.

Reply:
See #929